### ISSUE TITLE
DNX Beta-8 Publish HttpPlatformHandler cannot be found. This error is random

### ISSUE THREAD

**Issue by Tratcher** [aspnet/IISIntegration#32]
_From @CarlosFreire on November 10, 2015 21:11_

The description for Event ID 1001 from source HttpPlatformHandler cannot be found. Either the component that raises this event is not installed on your local computer or the installation is corrupted. You can install or repair the component on the local computer.

If the event originated on another computer, the display information had to be saved with the event.

The following information was included with the event: 

Process '4248' started successfully and is listening on port '12618'.

_Copied from original issue: aspnet/KestrelHttpServer#355_

**Comment by Tratcher** [comment]
Is something actually failing or are you just seeing odd output in your machine's event logs?  That message looks like what you'd get if you installed the VS beta8 tooling (which includes HttpPlatformHandler for IISExpress), but did not install the full HttpPlatformHandler for IIS.

**Comment by GuardRex** [comment]
I'm seeing this too in my logs; however, I can't trace it specifically to stopping and restarting AppPools (either for an old beta7-Helios app or a beta8-HttpPlatformHandler app). I'll poke around a bit more and see if I can figure out a pattern. btw- The apps seem to run fine AFAIK.

**Comment by GuardRex** [comment]
@Tratcher Ok ... I see when I'm getting this now. It's not with a manual stop/start of the AppPool, its with my little `msdeploy.bat` script that I hook from the VS PS publish script to publish with MSDeploy. The script looks like this ...
```
@echo off

set iis_website=<IIS Website Name>
set computer_name=https://<Azure VM>.cloudapp.net:<PORT>/msdeploy.axd
set username=<domain>\<username>
set password=<password>
set source_content_path=<path to VS publish output folder>
set dest_content_path=<path to app on server>
echo.
"C:\Program Files (x86)\IIS\Microsoft Web Deploy V3\msdeploy.exe" -verb:sync -allowUntrusted -source:recycleApp -dest:recycleApp="%iis_website%",recycleMode="StopAppPool",ComputerName="%computer_name%",UserName="%username%",Password="%password%",AuthType="Basic"
echo.
"C:\Program Files (x86)\IIS\Microsoft Web Deploy V3\msdeploy.exe" -verb:sync -allowUntrusted -source:contentPath="%source_content_path%" -dest:contentPath="%dest_content_path%",ComputerName="%computer_name%",UserName="%username%",Password="%password%",AuthType="Basic"
echo.
"C:\Program Files (x86)\IIS\Microsoft Web Deploy V3\msdeploy.exe" -verb:sync -allowUntrusted -source:recycleApp -dest:recycleApp="%iis_website%",recycleMode="StartAppPool",ComputerName="%computer_name%",UserName="%username%",Password="%password%",AuthType="Basic"
echo.
```
... and either that first (stop) command is causing it or the last (start) command is causing it. Yields ...
```
The description for Event ID 1001 from source HttpPlatformHandler cannot be found. Either the component that raises this event is not installed on your local computer or the installation is corrupted. You can install or repair the component on the local computer.

If the event originated on another computer, the display information had to be saved with the event.

The following information was included with the event: 

Process '2436' started successfully and is listening on port '31576'.
```
... however, the deployment succeeds and the app runs fine.

/cc @sayedihashimi

**Comment by sayedihashimi** [comment]
Did you see the doc at https://docs.asp.net/en/latest/publishing/iis.html?

**Comment by GuardRex** [comment]
@sayedihashimi ... You mean me? I am Luke.

**Comment by GuardRex** [comment]
@sayedihashimi I /cc'ed you because I thought if MSDeploy commands trigger this log entry, if MSDeploy makes it into a later version of VS publish, and something is amiss you might want to hear about it.

**Comment by ivanmartinvalle** [comment]
@GuardRex I can definitely reproduce this message without deploying. All I have to do is start the AppPool manually (after it has been stopped) AND hit the site.

Yeah, the site definitely runs fine with the message though. It's just the stopping of the AppPool that is causing me issues (the process is dying, as mentioned in #35 ).

**Comment by GuardRex** [comment]
@ivanmartinvalle Interesting ... I can't get it to log that way. Very interesting. Well ... at least it isn't breaking us. btw- What flavor of IIS are you on? My tests were with IIS 8 (WS 2012 ... not R2).

**Comment by ivanmartinvalle** [comment]
#35 describes my situation.

OS: Windows 7
IIS: 7.5.7600.16385

**Comment by sayedihashimi** [comment]
The HttpPlatformHandler requires IIS 8+ I believe.

**Comment by GuardRex** [comment]
We have Windows 2008 R2 (IIS 7.5) in the official doc. Perhaps that should be changed?

/cc @Rick-Anderson

**Comment by sayedihashimi** [comment]
Yes we may want to add a note that it currently requires IIS 8+

cc @Rick-Anderson

**Comment by Tratcher** [comment]
@sayedihashimi HttpPlatformHandler v1.2 is already working on Win7/R2. Some of the tooling may not have caught up yet though.

**Comment by sayedihashimi** [comment]
Ok sorry for the confusion then. Tooling shouldn't have any dep on IIS versions.

**Comment by GuardRex** [comment]
So just to recap where we are:

@CarlosFreire reported it but wasn't specific about a trigger.
@ivanmartinvalle has it happening after restarting the AppPool and hitting the site's endpoint.
I was just able to manually get it the same way (I didn't hit the endpoint the first time). The log is unremarkable ...
```
Hosting environment: Production
Now listening on: http://localhost:11735
Application started. Press Ctrl+C to shut down.
```

**Comment by Tratcher** [comment]
@rramachand21?

**Comment by Tratcher** [comment]
Note: The event log message "The description for Event ID 1001 from source HttpPlatformHandler cannot be found. Either the component that raises this event is not installed on your local computer or the installation is corrupted. You can install or repair the component on the local computer." may show up on machines that are using IIS Express without installing the HttpPlatformHandler msi. This is because Visual Studio contains its own copy of the HPH dll for use in IIS Express, it does not include the full installer.

**Comment by GuardRex** [comment]
The last instances I see of this in a couple of my logs are: In a 2012R2 log was 12/6. In a 2012 log was 1/6 ... actually just 20 minutes ago. The server-side cause is still under investigation?

**Comment by Tratcher** [comment]
This as been fixed in AspNetCoreModule v0.8: https://github.com/aspnet/IISIntegration/issues/105.
